lib/bolt/task.rb

```diff
@@ -77,6 +77,10 @@ def select_implementation(target, provided_features = [])
                impl = impls.find do |imp|
                  remote_impl = imp['remote']
                  remote_impl = metadata['remote'] if remote_impl.nil?
+                 unless imp['requirements'].is_a?(Array) || imp['requirements'].nil?
```

Comment:
Should this check also be done in Puppet? It seems like doing validation in one place would make this less confusing. This check seems like it would fit in the `find_implementations` method https://github.com/puppetlabs/puppet/blob/master/lib/puppet/module/task.rb#L133

Reply:
I had not thought of that. I guess that makes sense. If we do it in puppet I dont think we should do it in bolt (at least not have the tests set up the way they are because they will break as soon as we pull in a puppet version with the update). I'm not opposed to it.

Reply:
Yeah that sentence was ambiguous - 'also' = 'in addition to the other checks that are part of this issue', so we would only have the validation in Puppet. I think using the same pattern for both the `files` key check and the `requirements` makes it easier to find and understand how we validate metadata.

Reply:
Cool. Will raise new PR in puppet and close this.